homeassistant/components/bosch_alarm/coordinator.py

```diff
@@ -0,0 +1,76 @@
+"""Coordinator for Bosch Alarm Panel."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+import ssl
+
+from bosch_alarm_mode2 import Panel
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_MODEL, CONF_PASSWORD, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import CONF_INSTALLER_CODE, CONF_USER_CODE
+
+_LOGGER = logging.getLogger(__name__)
+
+type BoschAlarmConfigEntry = ConfigEntry[BoschAlarmCoordinator]
+
+
+class BoschAlarmCoordinator(DataUpdateCoordinator[Panel]):
```

Comment:
```suggestion
class BoschAlarmCoordinator(DataUpdateCoordinator[None]):
```
I'm not 100% sure, but if you set this to `None` you should be able to just leave out the _async_update method

Reply:
Unfortunately, `async_config_entry_first_refresh` always end up calling `_async_update_data` as it calls `async_refresh`
I wonder if it would make more sense to stub out `async_refresh` though since thats really the behaviour we don't have when we are push based.

Reply:
Do you have a preference of overriding `_async_update_data` vs overriding `_async_refresh`